rotkehlchen/chain/evm/decoding/pendle/utils.py

```diff
@@ -0,0 +1,104 @@
+            for market in response_data['markets']:
+                try:
+                    all_markets.append(deserialize_evm_address(market['address']))
+                    all_sy_tokens.append(deserialize_evm_address(market['sy'].split('-')[1]))
+                except (KeyError, ValueError) as e:
+                    log.warning(f'Skipping pendle market entry {market} due to {e!s}')
+                    continue
+
+    except (RemoteError, UnableToDecryptRemoteData, TypeError, KeyError) as e:
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+        log.error(f'Failed to retrieve Pendle markets for chain {chain} due to {msg}')
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.PENDLE_POOLS, str(chain.value)),
+            values=all_markets,
+        )
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.PENDLE_SY_TOKENS, str(chain.value)),
+            values=all_sy_tokens,
+        )
+
+
+def query_pendle_price(token: 'EvmToken') -> Price:
+    """Query USD price for a given Pendle PT/YT/SY/LP token"""
+    chain = token.chain_id
+    try:
+        result = request_get_dict(f'https://api-v2.pendle.finance/core/v1/{chain.value}/assets/prices?addresses={token.evm_address}')
+        return deserialize_price(result['prices'][to_normalized_address(token.evm_address)])
+    except (RemoteError, UnableToDecryptRemoteData, DeserializationError, TypeError, ValueError, KeyError) as e:  # noqa: E501
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+        log.error(f'Unable to query usd price of pendle wrapped token {token} due to {msg}')
+        return ZERO_PRICE
+
+
+def query_pendle_yield_tokens(evm_inquirer: 'EvmNodeInquirer') -> None:
+    """Queries all Pendle yield tokens(LP, SY, PT, YT) and saves them to allow price queries."""
+    all_tokens: set[ChecksumEvmAddress] = set()
+    try:
+        all_tokens.update(deserialize_evm_address(entry['address']) for entry in request_get_dict(f'https://api-v2.pendle.finance/core/v3/{evm_inquirer.chain_id.value}/assets/all'))
+    except (RemoteError, UnableToDecryptRemoteData, DeserializationError, KeyError) as e:
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!r}'
+        log.error(f'Unable to fetch pendle data for {evm_inquirer.chain_name} due to {msg}')
+        return
+
+    for token in all_tokens:
+        try:
+            get_or_create_evm_token(
+                evm_inquirer=evm_inquirer,
+                protocol=CPT_PENDLE,
+                userdb=evm_inquirer.database,
+                evm_address=token,
+                chain_id=evm_inquirer.chain_id,
+            )
```

Comment:
you are ignoring here the token. This function is quite heave so I would advice to use in combination with `get_token` that is much lighter. Once you have all the tokens in the DB it prevents from the heavy path